samples/iothub-connect-to-eventhubs/IotHubConnection.cs

```diff
@@ -0,0 +1,355 @@
+﻿
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Amqp.Sasl;
+using Microsoft.Azure.Amqp.Transport;
+
+namespace IotHubToEventHubsSample
+{
+    /// <summary>
+    ///   A transient connection to the IOT Hub service, providing a set of utility-type operations that
+    ///   span the service client and device roles.
+    /// </summary>
+    ///
+    public static class IotHubConnection
+    {
+        /// <summary>
+        ///   Requests connection string for the built-in IOT Hub messaging endpoint of the Event Hubs service.
+        /// </summary>
+        ///
+        /// <param name="iotHubConnectionString">The connection string for the IOT Hub instance to request the Event Hubs connection string from.</param>
+        /// <param name="timeout">The maximum amount of time that the request and translation should be allowed to take.  If not provided, a default of 60 seconds will be assumed.</param>
+        ///
+        /// <returns>A connection string which can be used to connect to the Event Hubs service and interact with the IOT Hub messaging endpoint.</returns>
+        ///
+        /// <exception cref="InvalidOperationException">The Event Hubs host information was not returned by the IOT Hub service.</exception>
+        ///
+        /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-endpoints" />
+        /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-read-builtin" />
+        /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-amqp-support#receive-telemetry-messages-service-client" />
+        ///
+        public static async Task<string> RequestEventHubsConnectionStringAsync(string iotHubConnectionString,
+                                                                               TimeSpan? timeout = default)
+        {
+            timeout ??= TimeSpan.FromMinutes(1);
+
+            if (string.IsNullOrEmpty(iotHubConnectionString))
+            {
+                throw new ArgumentException("The IOT Hub connection string must be provided.", nameof(iotHubConnectionString));
+            }
+
+            // Parse the connection string into the necessary components, and ensure the information
+            // is available.
+
+            var parsedTokens = new Dictionary<string, string>(ParseConnectionString(iotHubConnectionString), StringComparer.InvariantCultureIgnoreCase);
```

Comment:
Consider using https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.client.iothubconnectionstringbuilder.create?view=azure-dotnet#Microsoft_Azure_Devices_Client_IotHubConnectionStringBuilder_Create_System_String_ - which should reliably parse the iot connection string into parts.

Reply:
hrmm.... I wasn't aware that was a thing.   I'll take a look.

Reply:
Thanks for the suggestion!   That package was only compatible with the device-specific connection string, but the builder in `Microsoft.Azure.Devices` had a version that worked with the main `iothubowner` connection string.   Swapped that over.